distribution/pull_v2.go

```diff
@@ -346,10 +346,16 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named, platform
 		}
 		tagOrDigest = digested.Digest().String()
 	} else if tagged, isTagged := ref.(reference.NamedTagged); isTagged {
-		manifest, err = manSvc.Get(ctx, "", distribution.WithTag(tagged.Tag()))
+		tagService := p.repo.Tags(ctx)
+		desc, err := tagService.Get(ctx, tagged.Tag())
 		if err != nil {
 			return false, allowV1Fallback(err)
 		}
+
+		manifest, err = manSvc.Get(ctx, desc.Digest)
```

Comment:
Aren't there situations where Docker ends up storing the digest of the fetched image?  Would that make sense to check here in case it's not a manifest list?  (Does this function already do so? 😅)

Reply:
The switch on `manifest.(type)` happens just a little further down, if I'm understanding your question; if it's a manifest list at this point it will call the pullManifestList function

Reply:
Yeah, this is handled later on.